tests/functional/listing/test.js

```diff
@@ -0,0 +1,476 @@
+                });
+                done();
+            });
+    });
+    it('should list with big MaxKeys', done => {
+        s3.listObjects(
+            { Bucket,
+              MaxKeys: 15000,
+            },
+            (err, data) => {
+                assert.strictEqual(err, null);
+                cutAttributes(data);
+                assert.deepStrictEqual(data, {
+                    Marker: '',
+                    IsTruncated: false,
+                    Contents: [objects[0].Key,
+                               objects[1].Key,
+                               objects[2].Key,
+                               objects[3].Key,
+                               objects[4].Key,
+                               objects[5].Key,
+                               objects[6].Key,
+                               objects[7].Key,
+                               objects[8].Key,
+                               objects[9].Key,
+                              ],
+                    Name: Bucket,
+                    Prefix: '',
+                    MaxKeys: 15000,
+                    CommonPrefixes: [],
+                });
+                done();
+            });
+    });
+    it('should list with delimiter', done => {
+        s3.listObjects(
+            { Bucket,
+              Delimiter: '/',
+            },
+            (err, data) => {
+                assert.strictEqual(err, null);
+                cutAttributes(data);
+                assert.deepStrictEqual(data, {
+                    Marker: '',
+                    IsTruncated: false,
+                    Contents: [objects[0].Key],
+                    Name: Bucket,
+                    Prefix: '',
+                    Delimiter: '/',
+                    MaxKeys: 1000,
+                    CommonPrefixes: ['notes/'],
+                });
+                done();
+            });
+    });
+    it('should list with long delimiter', done => {
+        s3.listObjects(
+            { Bucket,
+              Delimiter: 'notes/summer',
+            },
+            (err, data) => {
+                assert.strictEqual(err, null);
+                cutAttributes(data);
+                assert.deepStrictEqual(data, {
+                    Marker: '',
+                    IsTruncated: false,
+                    Contents: [objects[0].Key,
+                               objects[1].Key,
+                               objects[2].Key,
+                               objects[3].Key,
+                               objects[7].Key,
+                               objects[8].Key,
+                               objects[9].Key,
+                              ],
+                    Name: Bucket,
+                    Prefix: '',
+                    Delimiter: 'notes/summer',
+                    MaxKeys: 1000,
+                    CommonPrefixes: ['notes/summer'],
+                });
+                done();
+            });
+    });
+    it('should list with delimiter and prefix related to #147', done => {
```

Comment:
copy-pasted this from the title of another test. I don't know what it means. by the way this number belongs to another repository and has nothing to do here